Docs - Persons (#8639)
PostHog/posthog.com: contents/docs/data/persons.mdx

    className="rounded shadow-xl"
/>

People in PostHog represent users who do events.
People in PostHog represent the users behind your events. You can view them in the [People tab](https://us.posthog.com/persons).

> Every event has a `distinct_id` representing a person attached to it. However, not every `distinct_id` has an associated person profile.

Reviewer: Ahh, I see you moved it down.

- Filter on user properties
- Create [cohorts](/docs/data/cohorts)

> - Use user properties for targeting feature flags, A/B tests, and surveys.

Reviewer: ```suggestion
- Target feature flags, A/B tests, and surveys.
```

Author: I will keep as is, since "Target feature flags, A/B tests, and surveys." makes it sound like you're not able to target at all with user properties (you can still target on other things)

> - Track initial UTM values and referrers across anonymous and [identified](/docs/product-analytics/identify) users.

Reviewer: Answering this: https://posthog.com/questions/select-users-in-sql

```suggestion
- Track initial UTM values and referrers across anonymous and [identified](/docs/product-analytics/identify) users.
- Query the `persons` table using [SQL insights](/docs/product-analytics/sql)
```

## People and person profiles

> Using people in PostHog requires capturing events with **person profiles**. Person profiles are how PostHog attributes events and properties to specific users. Examples of scenarios where you want to capture events with person profiles are:

Reviewer: All events are attributed to people, but only some people have person profiles.

- Tracking logged-in users in B2B and B2C SaaS apps

> - Low volume API server events

Reviewer: I don't full understand why low volume would change something, you could just not have very many users.

Reviewer: Alternatives:

- Growth and marketing teams wanting to analyze the complete conversion lifecycle
- Doing user segmented product analysis

> - Low volume API server events

Author: ```suggestion
- Doing user segmented product analysis
- Growth and marketing teams wanting to analyze the _complete_ conversion lifecycle
```

- B2C apps where users don't sign up or log in
- High volume API server events

> Under our current [pricing](/pricing), events with person profiles are about 4x more expensive than events without them (due to the compute power required to work with them), so it's best to capture them only when needed i.e., in the scenarios described above.

Reviewer: ```suggestion
Under our current [pricing](/pricing), events without person profiles can be up to 4x cheaper than events with them (due to computer cost of processing them), so it's best to capture them only when needed.
```

Reviewer: I think talking about how something can be cheaper is more appealing than how something could be more expensive.

})
```

> 

Reviewer: Should we have a link out to the user properties page here saying "Here's the docs on how to set user properties on person profiles"?

> ### Person profiles and distinct IDs
> 
> Below is good information to know when working with distinct IDs and person profiles:

Reviewer: ```suggestion
### More notes on person profiles and distinct IDs
```

- Every event has a `distinct_id` representing a user attached to it. However, not every `distinct_id` has an associated person profile.

- If a distinct ID already has a person profile created, any events captured with this distinct ID are considered events with a person profile. 

>     - To opt out of this behavior, [delete the person](#deleting-person-data) in the person details UI in PostHog or use the `DELETE` operation of the [person API endpoint](/docs/api/persons#delete-api-projects-project_id-persons-id).

Reviewer: We should address the not reusing `distinct_id` point: https://posthog.com/questions/why-do-i-see-person-not-found-for-a-user-that-gets-recreated-after-deletion (just realized this might be inconsistent)

Asked in Slack: https://posthog.slack.com/archives/C06R39WNY5B/p1717774463796589

> 

Reviewer: ```suggestion
```

})
```

### Person profiles and distinct IDs

Below is good information to know when working with distinct IDs and person profiles:

- Every event has a `distinct_id` representing a user attached to it. However, not every `distinct_id` has an associated person profile.

- If a distinct ID already has a person profile created, any events captured with this distinct ID are considered events with a person profile. 

>     - To opt out of this behavior, [delete the person](#deleting-person-data) in the person details UI in PostHog or use the `DELETE` operation of the [person API endpoint](/docs/api/persons#delete-api-projects-project_id-persons-id).

Author: ```suggestion
    - To opt out of this behavior, contact file a support ticket via the [in-app modal](https://app.posthog.com/home#supportModal).
```
